CodeEdit/Features/CodeEditUI/Views/IconToggleStyle.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  IconToggleStyle.swift
+//  CodeEdit
+//
+//  Created by Austin Condiff on 11/9/23.
+//
+
+import SwiftUI
+
+struct IconToggleStyle: ToggleStyle {
+    var font: Font?
+    var size: CGSize?
+
+    @State var isPressing = false
+
+    init() {
+        self.font = nil
+        self.size = nil
+    }
+
+    init(font: Font? = nil, size: CGFloat? = nil) {
+        self.font = font
+        self.size = size == nil ? nil : CGSize(width: size ?? 0, height: size ?? 0)
+    }
+
+    init(font: Font? = nil, size: CGSize? = nil) {
+        self.font = font
+        self.size = size
+    }
+
+    init(font: Font? = nil) {
+        self.font = font
+        self.size = nil
+    }
```

Comment:
Because you're using default parameters, you can combine all these into two functions

```swift
init(font: Font? = nil, size: CGFloat? = nil) {
    self.font = font
    if let size = size {
        self.size = CGSize(width: size, height: size)
    } else {
        self.size = nil
    }
}

init(font: Font? = nil, size: CGSize? = nil) {
    self.font = font
    self.size = size
}
```

Reply:
I do that and I get this error...
<img width="706" alt="image" src="https://github.com/CodeEditApp/CodeEdit/assets/806104/5e3f39ed-b43c-47b5-819a-a38d723a6349">

Reply:
We can also remove this function from here too:
```swift
init() {
        self.font = nil
        self.size = nil
}
```